contrib/inventory/azure_rm.py

```diff
@@ -687,39 +727,43 @@ def _load_machines(self, machines):
-                    if self.group_by_security_group and \
-                       self._security_groups[resource_group].get(network_interface.id, None):
-                        host_vars['security_group'] = \
-                            self._security_groups[resource_group][network_interface.id]['name']
-                        host_vars['security_group_id'] = \
-                            self._security_groups[resource_group][network_interface.id]['id']
-                    host_vars['network_interface'] = network_interface.name
-                    host_vars['network_interface_id'] = network_interface.id
-                    host_vars['mac_address'] = network_interface.mac_address
-                    for ip_config in network_interface.ip_configurations:
-                        host_vars['private_ip'] = ip_config.private_ip_address
-                        host_vars['private_ip_alloc_method'] = ip_config.private_ip_allocation_method
-                        if self.use_private_ip:
-                            host_vars['ansible_host'] = ip_config.private_ip_address
-                        if ip_config.public_ip_address:
-                            public_ip_reference = self._parse_ref_id(ip_config.public_ip_address.id)
-                            public_ip_address = self._network_client.public_ip_addresses.get(
-                                public_ip_reference['resourceGroups'],
-                                public_ip_reference['publicIPAddresses'])
-                            if not self.use_private_ip:
-                                host_vars['ansible_host'] = public_ip_address.ip_address
-                            host_vars['public_ip'] = public_ip_address.ip_address
-                            host_vars['public_ip_name'] = public_ip_address.name
-                            host_vars['public_ip_alloc_method'] = public_ip_address.public_ip_allocation_method
-                            host_vars['public_ip_id'] = public_ip_address.id
-                            if public_ip_address.dns_settings:
-                                host_vars['fqdn'] = public_ip_address.dns_settings.fqdn
+            # here we are assuming all vmss vms have private ip addresses
+            if vmss_vm:
+                host_vars['ansible_host'] = machine.private_ip_address
+            else:
+                for interface in machine.network_profile.network_interfaces:
+                    interface_reference = self._parse_ref_id(interface.id)
+                    network_interface = self._network_client.network_interfaces.get(
+                        interface_reference['resourceGroups'],
+                        interface_reference['networkInterfaces'])
+                    if network_interface.primary:
+                        if self.group_by_security_group and \
+                           self._security_groups[resource_group].get(network_interface.id, None):
+                            host_vars['security_group'] = \
```

Comment:
get security group info from vnet

Reply:
I cannot seem to find a way to associate the network_interface with the subnet . For example here is an id:
`/subscriptions/redacted/resourceGroups/test/providers/Microsoft.Compute/virtualMachineScaleSets/testvmss/virtualMachines/0/networkInterfaces/testvmss-nic'`

The virtual machine profile inside the vmss, returns None for network security groups :(

I may have missed something here.

Reply:
you can get subnet info from `self.compute_client.virtual_machine_scale_sets.get(self.resource_group, self.name)`